### PR TITLE
Re-enable io.js/node 0.12 build

### PR DESCRIPTION
**Pull request by simov**
Fixed the negative msecs error and re-enabled io.js and node 0.12 in the build.

Closes https://github.com/request/request/issues/1421
Closes https://github.com/request/request/issues/1414

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,7 +41,7 @@
     "isstream": "~0.1.1"
   },
   "scripts": {
-    "test": "npm run lint && node node_modules/.bin/taper tests/test-*.js && npm run test-browser && npm run clean",
+    "test": "npm run lint && node node_modules/.bin/taper tests/test-*.js && npm run test-browser",
```
Comment (nicolasmccurdy) [inline package.json]: Out of curiosity, why was this removed?

**Comment by nicolasmccurdy** [comment]
:+1: Nice job!

File: package.json (hunk 1 above)
Comment (simov) [inline package.json]: I think we had a clean script back then, that logic was later moved to the browser test itself. NPM on node 0.10 doesn't complain about the missing script, where the newer version of it throws an error.

Comment (nicolasmccurdy) [inline package.json]: Ah, it seems like this is safe to remove then. npm won't look anywhere else for scripts to run (see [run](https://docs.npmjs.com/cli/run-script)).

Comment (eiriksm) [inline package.json]: Yes. I think the clean script removed a browserified build of the test script, but the current version browserifies the test script on the fly instead.
